Add DiscoveryInfo to task

Vlad777 opened the pull request:
Can someone take a look at how this should be done?
We'd like to be able to set a DNS record in mesos for the memsql master.
Ideally, it should also include the cluster name in the DNS, which would need to be exposed as a label in DiscoveryInfo perhaps?
Currently, there will be an issue if the framework starts more than one cluster.

wsong commented:
To be honest, I'm not very familiar with this feature myself.  As you can see, the cluster name is available as cluster_name in the function you're modifying.

You can build your own scheduler Docker images in this repository by running `make build-scheduler`; after you've built such an image, you can load it onto your own DCOS cluster to test it out.  It's a bit of a pain, but it might be worth investigating:

https://docs.mesosphere.com/dev-service/overview/

That way, you could test out your own features and you wouldn't have to rely on us pushing changes to Docker Hub in order to try them out.

Vlad777 commented:
Ok, I've verified that the changes I made work in our environment, including our own backend scripts that use the DiscoveryInfo data to update DNS.
I figured out that the cluster_name being passed is really the cluster_id, so I added the display_name.
I also added the host_ip that I want for the DNS resolution.

Vlad777 commented:
Any issues with merging this?
